lib/modules/manager/pub/artifacts.ts

```diff
@@ -49,10 +48,21 @@ export async function updateArtifacts({
       const depNames = updatedDeps
         .map((dep) => dep.depName)
         .filter(is.string)
-        .filter((depName) => depName !== 'flutter')
-        .map(quote)
-        .join(' ');
-      cmd.push(`${toolName} pub upgrade ${depNames}`);
+        .map(quote);
+
+      if (depNames.length === 1 && SDK_NAMES.includes(depNames[0])) {
+        cmd.push(`${toolName} pub get`);
```

Comment:
what does this command do?

Reply:
`dart pub get` / `flutter pub get` are essentially the `yarn install` command. We can't use the `pub upgrade dart` command because that's for updating dependencies, not the SDK. 

So using the `pub get` command instead, which would refresh the lock file with the new SDK constraint as well as updating any transitive dependencies required by the SDK.

Reply:
ok, I think you should add the `--no-precompile` flag to safe runtime

https://dart.dev/tools/pub/cmd/pub-get#--no-precompile